### PR TITLE
Slack new commands

### PR DESCRIPTION
**Pull request by liorblob**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/18455
fixes: https://github.com/demisto/etc/issues/13803

## Description
Added the following commands:
* slack-close-channel (now with optional `channel` argument)
* slack-create-channel
* slack-invite-to-channel
* slack-kick-from-channel
* slack-rename-channel
* slack-get-user-details

## Required version of Demisto
5.0

## Does it break backward compatibility?
   - No

## Must have
- [X] Tests
- [X] Documentation https://github.com/demisto/etc/issues/18935
- [X] Code Review

### CODE REVIEW

**Comment by liorblob** [comment]
@kirbles19 Please review `Slack.py`, `Slack.yml` in accordance with the new commands above.

File: Integrations/Slack/Slack.py [hunk 1]
```diff
@@ -114,6 +117,31 @@ def get_user_by_name(user_to_search: str, integration_context: dict) -> dict:
     return user
 
 
+def search_slack_users(integration_context, users):
```
Comment (Itay4) [inline Integrations/Slack/Slack.py]: please add types

File: Integrations/Slack/Slack.py [hunk 2]
```diff
@@ -217,6 +245,21 @@ def invite_users_to_conversation(conversation_id: str, users_to_invite: list):
                 raise
 
 
+def kick_users_from_conversation(conversation_id: str, users_to_invite: list):
```
Comment (Itay4) [inline Integrations/Slack/Slack.py]:
```suggestion
def kick_users_from_conversation(conversation_id: str, users_to_kick: list):
```

File: Integrations/Slack/Slack.yml [hunk 3]
```diff
@@ -339,6 +375,115 @@ script:
...
+      secret: false
+    - default: false
+      description: The channel name. If not specified, the mirrored investigation
+        channel is used (if the channel exists).
+      isArray: false
+      name: channel
+      required: false
+      secret: false
+    deprecated: false
+    description: KIck a user from a channel in Slack.
```
Comment (Itay4) [inline Integrations/Slack/Slack.yml]:
```suggestion
    description: Kick one or more users from a channel in Slack.
```

File: Integrations/Slack/Slack.yml [hunk 4]
```diff
@@ -339,6 +375,115 @@ script:
     description: Sets the topic for a channel.
     execution: false
     name: slack-set-channel-topic
+  - arguments:
```
Comment (Itay4) [inline Integrations/Slack/Slack.yml]: lets add topic argument

File: Integrations/Slack/Slack.yml [hunk 5]
```diff
@@ -339,6 +375,115 @@ script:
...
+      name: users
+      required: false
+      secret: false
+    deprecated: false
+    description: Create a channel in Slack.
+    execution: false
+    name: slack-create-channel
+  - arguments:
+    - default: true
+      description: CSV list of the users to invite.
```
Comment (Itay4) [inline Integrations/Slack/Slack.yml]: please add example

File: Integrations/Slack/Slack.yml [hunk 6]
```diff
@@ -339,6 +375,115 @@ script:
...
+      name: channel
+      required: false
+      secret: false
+    deprecated: false
+    description: Invite a user to a channel in Slack.
+    execution: false
+    name: slack-invite-to-channel
+  - arguments:
+    - default: true
+      description: CSV list of users to kick.
```
Comment (Itay4) [inline Integrations/Slack/Slack.yml]: please add example

File: Integrations/Slack/Slack.yml [hunk 7]
```diff
@@ -339,6 +375,115 @@ script:
...
+      name: channel
+      required: false
+      secret: false
+    deprecated: false
+    description: Rename a channel in Slack.
+    execution: false
+    name: slack-rename-channel
+  - arguments:
+    - default: true
+      description: The Slack user.
```
Comment (Itay4) [inline Integrations/Slack/Slack.yml]: is it the username? display name?

File: Integrations/Slack/Slack.yml [hunk 8]
```diff
@@ -339,6 +375,115 @@ script:
...
+    - contextPath: Slack.User.Name
+      description: The user's name.
+      type: String
+    - contextPath: Slack.User.DisplayName
+      description: The user's display name.
+      type: String
+    - contextPath: Slack.User.Email
+      description: The user's email.
+      type: String
   dockerimage: demisto/slack:1.0.0.814
```
Comment (Itay4) [inline Integrations/Slack/Slack.yml]:
```suggestion
  dockerimage: demisto/slack:1.0.0.1560
```

File: Integrations/Slack/Slack_test.py [hunk 9]
```diff
@@ -2079,54 +2149,239 @@ def test_send_topic_no_args_no_investigation(mocker):
...
+
+    # Assert
+    assert Slack.get_conversation_by_name.call_count == 1
+    assert Slack.invite_users_to_conversation.call_count == 1
+    assert success_results[0] == 'Successfully invited users to the channel.'
+    assert send_args[0] == 'C012AB3CD'
+    assert send_args[1] == ['U012A3CDE', 'U07QCRPA4']
+
+
+def test_invite_users_no_channel(mocker):
```
Comment (Itay4) [inline Integrations/Slack/Slack_test.py]: lets add test to invite users, with no channel given and no mirror channel is found, and assert the error

File: TestPlaybooks/playbook-Slack_TestPlaybook.yml [hunk 10]
```diff
@@ -323,12 +329,180 @@ tasks:
     note: false
     timertriggers: []
     ignoreworker: false
+  "10":
```
Comment (Itay4) [inline TestPlaybooks/playbook-Slack_TestPlaybook.yml]: can we add the invite user, kick user, rename channel and get user commands to the test playbook ?

**Comment by ygerowitz** [comment]
Updated the Slack.yml file.

**Comment by liorblob** [comment]
@ygerowitz @kirbles19 I made some adjustments to the descriptions.

**Comment by kirbles19** [comment]
@liorblob Done.

**Pull request merged**